File: Wayne/WayneCompiler/src/Dfir_representation/InputNode.java
package Dfir_representation;

public class InputNode{
	   private int parentId;
	   private int nodeId;
	   public InputNode(int NodeId,int parentId){
		   this.parentId = parentId;
		   this.nodeId = NodeId;
	   }
	   public void print(){
		   System.out.println("NodeId: " + this.nodeId + ' ' + "ParentId: " + this.parentId);
	   }
	public int getParentId() {
		return parentId;
	}
	public void setParentId(int parentId) {
		this.parentId = parentId;
	}
	public int getNodeId() {
		return nodeId;
	}
	public void setNodeId(int nodeId) {
		this.nodeId = nodeId;
	}
}

File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/InputNode.cs
using System;

namespace DFIR_Compiler
{
    public class InputNode
    {
        private int parentId;
        private int nodeId;
        public InputNode(int NodeId, int ParentId)
        {
            nodeId = NodeId;
            parentId = ParentId;            
        }

        public virtual int GetNodeId()
        {
            return nodeId;
        }
        public virtual int GetParentId()
        {
            return parentId;
        }


        public virtual void SetNodeId(int NodeId)
        {
            nodeId = NodeId;
        }
        public virtual void SetParentId(int parentId)
        {
            this.parentId = parentId;
        }


        public virtual void Print()
        {
            Console.WriteLine("InputNode:");
            Console.WriteLine("NodeId: " + nodeId + ' ' + "ParentId: " + parentId);
        }
    }
}
